### PR TITLE
For #144 clear and update packages count in meta files

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #144 and #137: 
- extracted class `XmlAlter` to alter packages count if metadata files
- made `XmlPrimaryMaid` implement `XmlMaid`
- added new methods to `FileOutput`
- extracted interface `Metadata` from `MetadataFile`
- fully implemented `ModifiableMetadata`

This PR is large, I'm sorry, but in my defense it does not have any new code, it's reorganizing.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#150](https://github.com/artipie/rpm-adapter/issues/150) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/4df56db3-5d48-4420-bd20-d552815cb96b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/rpm/ModifiableRepository.java [hunk 1]
```diff
@@ -97,9 +97,12 @@ public ModifiableRepository update(final FilePackage pkg) throws IOException {
      * Clears records about packages that does not present in the repository any more
      * from metadata files.
      * @return Itself
+     * @throws IOException On error
      */
-    public ModifiableRepository clear() {
-        this.metadata.forEach(item -> item.clear(this.existing));
+    public ModifiableRepository clear() throws IOException {
+        for (final Metadata item : this.metadata) {
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/ModifiableRepository.java:103]: @olenagerasimova does this metadata items shares some common state? If not, I'd use:
```java
this.metadata().stream().parallel().forEach(Metadata::brash)
```
Parallel update for metadata items increases performance of RPM more than twice.

File: src/main/java/com/artipie/rpm/pkg/Metadata.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * @since 0.8
+ */
+public interface Metadata extends PackageOutput {
+
+    /**
+     * Brushes metadata by cleaning not existing packages and setting packages count.
+     * @param ids Ids of the packages to clear
+     * @throws IOException When error occurs
+     */
+    void brash(List<String> ids) throws IOException;
```
Comment (g4s8) [inline src/main/java/com/artipie/rpm/pkg/Metadata.java:44]: @olenagerasimova maybe brush?

**Comment by olenagerasimova** [comment]
@g4s8 thanks, updated

**Comment by olenagerasimova** [comment]
@g4s8 ping

**Comment by 0crat** [comment]
@vatavuk[/z](https://www.0crat.com/u/vatavuk) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by Vatavuk** [comment]
@g4s8 please remove this issue from the scope

**Comment by 0crat** [comment]
@vatavuk[/z](https://www.0crat.com/u/vatavuk) resigned from [#150](https://github.com/artipie/rpm-adapter/issues/150), since the job is not in scope anymore

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@da945c9f-54c4-44f9-95c4-928185ec3345, thread: PQ-CT2E6TK9B
com.zerocracy.claims.ClaimIn[214] java.lang.IllegalArgumentException: Parameter "age" not found in "Order was canceled" among: [reason, priority, job, cause, _parent_sig, flow, voluntarily, login]
```

[0.53.14](https://github.com/zerocracy/farm): CID: [ebb91fdb-f8c9-4e73-9084-6d09cb0017e1](https://www.0crat.com/footprint/CT2E6TK9B/ebb91fdb-f8c9-4e73-9084-6d09cb0017e1), Type: "Order was canceled"

**Comment by Vatavuk** [comment]
@g4s8 mr. Zerocrat refused cooperation :)

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/rpm-adapter/issues/150#issuecomment-624535907))

@g4s8 This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#150](https://github.com/artipie/rpm-adapter/issues/150) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Frpm-adapter%23150%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Pull request merged**